integration/hive/hive-openlineage-hook/src/main/java/io/openlineage/hive/client/EventEmitter.java

```diff
@@ -0,0 +1,61 @@
+/*
+/* Copyright 2018-2025 contributors to the OpenLineage project
+/* SPDX-License-Identifier: Apache-2.0
+*/
+package io.openlineage.hive.client;
+
+import static com.google.common.hash.Hashing.sha512;
+import static io.openlineage.client.utils.UUIDUtils.generateNewUUID;
+
+import io.openlineage.client.Clients;
+import io.openlineage.client.OpenLineage.RunEvent;
+import io.openlineage.client.OpenLineageClient;
+import io.openlineage.client.OpenLineageClientException;
+import io.openlineage.client.OpenLineageClientUtils;
+import io.openlineage.hive.api.OpenLineageContext;
+import io.openlineage.hive.util.NetworkUtils;
+import java.util.UUID;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.hadoop.conf.Configuration;
+
+@Getter
+@Slf4j
+public class EventEmitter implements AutoCloseable {
+  private final OpenLineageClient client;
+  private final UUID runId;
+  private final String jobName;
+  private final String jobNamespace;
+
+  public EventEmitter(OpenLineageContext olContext) {
+    Configuration conf = olContext.getHadoopConf();
+    this.client = Clients.newClient(olContext.getOpenLineageConfig());
+    this.runId = generateNewUUID();
+    this.jobNamespace =
+        conf.get(
+            HiveOpenLineageConfigParser.NAMESPACE_KEY, NetworkUtils.LOCAL_IP_ADDRESS.getHostName());
+    this.jobName =
+        conf.get(
+            HiveOpenLineageConfigParser.JOB_NAME_KEY, getJobNamespace(olContext.getQueryString()));
+  }
+
+  public void emit(RunEvent event) {
+    try {
+      client.emit(event);
+      log.debug(
+          "Emitting lineage completed successfully: {}", OpenLineageClientUtils.toJson(event));
+    } catch (OpenLineageClientException exception) {
+      log.error("Could not emit lineage", exception);
+    }
+  }
+
+  public static String getJobNamespace(String queryString) {
+    // TODO: Confirm that this is an appropriate hashing function to use
+    return sha512().hashUnencodedChars(queryString).toString();
```

Comment:
Spark command generates OpenLineage events with jobName like `insert_hadoop_fs_relation` or `insert_hadoop_fs_relation.mydb.mytable`. Maybe here we can use same approach, e.g. generate default job names like `insert_into.mydb.mytable`

Reply:
HookContext has `getOperationName()` method:
* https://github.com/apache/hive/blob/rel/release-3.1.3/ql/src/java/org/apache/hadoop/hive/ql/hooks/HookContext.java
* https://github.com/apache/hive/blob/rel/release-3.1.3/ql/src/java/org/apache/hadoop/hive/ql/plan/HiveOperation.java

Reply:
Yes, random hash seems bad for a namespace. Job namespace should be consistent between job runs.
I would 

- look for some override (like configured namespace, or `OPENLINEAGE_NAMESPACE` env var)
- have some good default like ⬆️ 
- if there's no good default, use something like `default`.

Reply:
@mobuchowski this method already is used only if the name is not defined in config 

@mobuchowski @dolfinus the operation names we now have are `QUERY` or `CREATETABLE_AS_SELECT`, we could use those 
while in case of CTAS we could extract name of output from lineage we collect. I'm not sure about this in case of query, maybe we could use first input dataset name?

Reply:
> I'm not sure about this in case of query, maybe we could use first input dataset name?

Could you please give an example of QUERY? Is it a simple SELECT which returns data to uset without writing it, or ...?

I'm not insisting on adding dataset name as a mandatory part of default job name. It's just allows to avoid confusion when 2 completely independent queries use the same job name, and some OpenLineage implementations (like Marquez) group all events by job name, and these events become 2 parts of the same lineage tree which is not the case.

Reply:
the default `namespace` is `default`, the default `name` will be `operationName`.`database`.`table`

> Could you please give an example of QUERY? Is it a simple SELECT which returns data to uset without writing it, or ...?

all select queries but I forgot that we check for existing output datasets so its no issue here